src/main/java/com/tfgm/persistence/TramNetworkDTORepoImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.tfgm.persistence;
+
+import com.tfgm.models.Tram;
+import com.tfgm.models.TramNetworkDTO;
+import com.tfgm.models.TramStop;
+import com.tfgm.models.TramStopContainer;
+import com.tfgm.persistence.mapper.TramNetworkDTOMapper;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class TramNetworkDTORepoImpl implements TramNetworkDTORepo {
+
+  private final TramNetworkDTOMapper TNMapper;
+
+  public TramNetworkDTORepoImpl(TramNetworkDTOMapper TNMapper) {
+    this.TNMapper = TNMapper;
+  }
+
+  @Override
+  public List<TramNetworkDTO> getAll() {
+    return TNMapper.getAll();
+  }
+
+    @Override
+    public List<Long> getAllTimestamps() {
+        return TNMapper.getAllTimestamps();
+    }
+
+    @Override
+  public void saveTramNetwork(Map<String, TramStop> tramStopHashMap) {
+
+    List<Tram> allTrams = new ArrayList<>();
+
+    for (TramStop tramStop : tramStopHashMap.values()) {
```

Comment:
This looks much better. Maybe consider just adding helper methods to extract the logic out to avoid nested loops etc, unless there is an approach that would avoid the nesting altogether.

Reply:
That makes sense, unfortunately nested loops are unavoidable as I need to get evert Tram from the queues between every tramstop